modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/details.jsp

```diff
@@ -200,31 +148,33 @@ if (parentGroupId != GroupConstants.DEFAULT_PARENT_GROUP_ID) {
 		<aui:input inlineLabel="right" label="limit-membership-to-members-of-the-parent-site" labelCssClass="simple-toggle-switch" name="membershipRestriction" type="toggle-switch" value="<%= membershipRestriction %>" />
 	</div>
 
-	<aui:script use="liferay-search-container">
-		A.one('#<portlet:namespace />selectParentSiteLink').on('click', (event) => {
-			Liferay.Util.openSelectionModal({
-				onSelect: function (event) {
-					var searchContainer = Liferay.SearchContainer.get(
-						'<portlet:namespace />parentGroupSearchContainer'
-					);
+	<aui:select label="membership-type" name="type">
+		<aui:option label="open" value="<%= GroupConstants.TYPE_SITE_OPEN %>" />
+		<aui:option label="restricted" value="<%= GroupConstants.TYPE_SITE_RESTRICTED %>" />
+		<aui:option label="private" value="<%= GroupConstants.TYPE_SITE_PRIVATE %>" />
+	</aui:select>
 
-					var rowColumns = [];
+	<%
+	boolean manualMembership = true;
 
-					var href =
-						'<portlet:renderURL><portlet:param name="mvcPath" value="/edit_site.jsp" /><portlet:param name="redirect" value="<%= currentURL %>" /></portlet:renderURL>&<portlet:namespace />groupId=' +
-						event.entityid;
+	if (liveGroup != null) {
+		manualMembership = GetterUtil.getBoolean(liveGroup.isManualMembership(), true);
+	}
+	%>
 
-					rowColumns.push(event.entityname);
-					rowColumns.push(event.grouptype);
-					rowColumns.push(
-						'<a class="modify-link" data-rowId="' +
-							event.entityid +
-							'" href="javascript:;"><%= UnicodeFormatter.toString(removeGroupIcon) %></a>'
```

Comment:
This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation: